src/test/java/io/confluent/connect/jdbc/source/integration/JdbcSourceConnectorIT.java

```diff
@@ -0,0 +1,339 @@
+    connect.deleteConnector(CONNECTOR_NAME);
+
+// start a source connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for tasks to spin up
+    int expected = Integer.valueOf(MAX_TASKS); // or set to actual number
+    waitForConnectorToStart(CONNECTOR_NAME, expected);
+
+    Thread.sleep(3000);
+
+    ConsumerRecords<byte[], byte[]> newRecords = connect.kafka().consume(
+            NUM_RECORDS,
+            CONSUME_MAX_DURATION_MS,
+            "sql-mysqlTable");
+    Assert.assertEquals(NUM_RECORDS, newRecords.count());
+    sendTestDataToMysql(start);
+    Thread.sleep(TimeUnit.SECONDS.toMillis(3));
+    ConsumerRecords<byte[], byte[]> newestRecords = connect.kafka().consume(
+            NUM_RECORDS*2,
+            CONSUME_MAX_DURATION_MS,
+            "sql-mysqlTable");
+  }
+
+  @Test
+  public void testConfigUpdate() throws InterruptedException, SQLException {
+    int start = 0;
+    connect.kafka().createTopic("sql-" + KAFKA_TOPIC);
+    dropTableIfExists(KAFKA_TOPIC);
+    createTimestampTable();
+    sendTestTimestampDataToMysql(start);
+    // Add mysql dialect related configurations.
+    props.put("connection.password", "password");
+    props.put("connection.user", "user");
+    props.put("connection.url", "jdbc:mysql://localhost:3307/db");
+    props.put("dialect.name", "MySqlDatabaseDialect");
+    props.put("value.converter", JsonConverter.class.getName());
+    props.put("mode", "incrementing");
+    props.put("incrementing.column.name",  "id");
+    props.put("topic.prefix",  "sql-");
+
+    // start a source connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for tasks to spin up
+    int expectedNumTasks = Integer.valueOf(MAX_TASKS); // or set to actual number
+    waitForConnectorToStart(CONNECTOR_NAME, expectedNumTasks);
+
+    Thread.sleep(3000);
+
+    log.info("Waiting for records in destination topic ...");
+    ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(
+            NUM_RECORDS,
+            CONSUME_MAX_DURATION_MS,
+            "sql-mysqlTable");
+    Assert.assertEquals(NUM_RECORDS, records.count());
+    connect.deleteConnector(CONNECTOR_NAME);
+
+    props.put("mode", "timestamp");
+    props.put("timestamp.column.name",  "timeof");
+    props.put("topic.prefix",  "sql-");
+
+    sendTestTimestampDataToMysql(start+NUM_RECORDS);
+
+
+    // start a source connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for tasks to spin up
+    int expected = Integer.valueOf(MAX_TASKS); // or set to actual number
+    waitForConnectorToStart(CONNECTOR_NAME, expected);
+
+    Thread.sleep(3000);
+
+    ConsumerRecords<byte[], byte[]> newRecords = connect.kafka().consume(
+            NUM_RECORDS,
+            CONSUME_MAX_DURATION_MS,
+            "sql-mysqlTable");
+
+    Assert.assertEquals(NUM_RECORDS*2, newRecords.count());
+
+  }
+
+
+  private void dropTableIfExists(String kafkaTopic) throws SQLException {
```

Comment:
These functions can be moved to BaseIT for a neat implementation.
Just have tests in this class.